README.md

```diff
@@ -172,12 +191,15 @@ go about writing custom health checks.
 
 ### Writing Custom Health Checkers
 
-Every APi and application is different. The kind of metrics you may need to track
-can be different from what others do. This module is all about being flexible,
-while being designed for consistency and RFC-compliance. Adding custom health 
-checks is very easy. All you need to do is to call an `addCheck()` call with
-the name of the component and metrics that the check is related to, as well
-as an ES2018 async function that will be executed to retrieve the health values.
+Every APi and application is different. The kind of metrics you may need to
+track can be different from what others do. This module is all about being
+flexible, while being designed for consistency and RFC-compliance. Adding custom
+health checks is very easy. All you need to do is to call an `addCheck()` method
+with the name of the component and metrics that the check is related to, as well
+as pass an ES2018 async function that will be executed to retrieve the health
+values. Because some existing libraries produce promises, to free you from
```

Comment:
I might be missing something, here, but I don't see how this ever makes sense.  Promises do not resolve with new values (they aren't observable streams or anything), so such a "healthcheck" would only check once, and then implicitly cache the result forever.

I think in order to get the effect of allowing anything that returns a promise, you can just allow any function, and disallow promises.  In the execution of the check:

```
resp.promises = currRun.map(async check => check());
```

gives a list of promises.  If the `check` returned a promise, then that will be the result, and if it did not, the `async` will wrap the result in a promise.

Reply:
Great point! You are spot! Fixed by: a37a955

The actual code is `resp.promises = currRun.map(async check => check.executor());` ", but the idea is the important thing. /tomlehrer" :)

Also updated documentation, unit-tests and examples.